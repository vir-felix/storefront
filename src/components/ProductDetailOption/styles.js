export default (theme) => ({
  variantOptionContainer: {
    display: "flex",
    width: `${theme.spacing.unit * 3.125}%`,
    maxWidth: `${theme.spacing.unit * 3.125}%`,
    padding: theme.spacing.unit * 1.25
  },
  optionButton: {
    "flex": "1 1 auto",
    "fontWeight": 600,
    "padding": theme.spacing.unit * 1.25,
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.borderColor,
    "&:hover": {
      backgroundColor: theme.palette.reaction.activeElementBackground
    }
  },
  optionText: {
    fontWeight: 500,
    fontSize: "14px"
  },
  isSelected: {
    backgroundColor: theme.palette.reaction.activeElementBackground
  }
});
