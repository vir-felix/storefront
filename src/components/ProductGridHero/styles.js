export default (theme) => ({
  heroImg: {
    width: "100%",
    height: "325px",
    objectFit: "cover"
  },
  productGridContainer: {
    maxWidth: theme.grid.productGridMaxWidth,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
