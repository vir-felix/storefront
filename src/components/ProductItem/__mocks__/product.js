export default {
  product: {
    _id: "cmVhY3Rpb24vY2F0YWxvZ1Byb2R1Y3Q6am5hOTVrYTlhTmtNeXRteGg=",
    title: "Men's Lightweigth Synchilla",
    slug: "men's-lightweigth-synchilla",
    description: "Great jacket!",
    vendor: "Patagonia",
    isLowQuantity: false,
    isSoldOut: true,
    isBackorder: false,
    primaryImage: {
      toGrid: 1,
      priority: 0,
      productId: "cmVhY3Rpb24vcHJvZHVjdDpqbmE5NWthOWFOa015dG14aA==",
      URLs: {
        thumbnail: "/assets/files/Media/aHEWna7yeBvNFM9bL/thumbnail/25580_CICN.png",
        small: "/assets/files/Media/aHEWna7yeBvNFM9bL/small/25580_CICN.png",
        medium: "/assets/files/Media/aHEWna7yeBvNFM9bL/medium/25580_CICN.jpg",
        large: "/assets/files/Media/aHEWna7yeBvNFM9bL/large/25580_CICN.jpg"
      }
    },
    price: {
      min: 240,
      max: 240
    }
  },
  positions: [
    {
      displayWeight: 0
    }
  ]
};
