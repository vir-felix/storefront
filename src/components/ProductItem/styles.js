export const styles = (theme) => ({
  "@keyframes spin": {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(-360deg)" }
  },
  "root": {},
  "productInfo": {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    paddingTop: 10
  },
  "productMedia": {
    backgroundColor: theme.palette.primary.contrastText,
    position: "relative"
  },
  "img": {
    height: "auto",
    width: "100%"
  },
  "imgLoading": {
    alignItems: "center",
    backgroundColor: "transparent",
    display: "flex",
    fontSize: 48,
    height: 100,
    justifyContent: "center",
    left: "calc(50% - 50px)",
    position: "absolute",
    top: "calc(50% - 50px)",
    width: 100
  },
  "loadingIcon": {
    color: theme.palette.primary.light,
    fontSize: "inherit",
    animationName: "spin",
    animationDuration: theme.transitions.duration.standard * 2,
    animationTimingFunction: theme.transitions.easing.sharp,
    animationIterationCount: "infinite",
    animationFillMode: "both"
  },
  "link": {
    ...theme.typography.body2,
    "textAlign": "left",
    "&:hover": {
      color: theme.palette.secondary.main
    }
  }
});
