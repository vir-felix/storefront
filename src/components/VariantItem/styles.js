export default (theme) => ({
  variantButton: {
    "display": "flex",
    "justifyContent": "space-between",
    "textTransform": "none",
    "width": "100%",
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.contrastText,
    "paddingTop": theme.spacing.unit * 1.25,
    "paddingBottom": theme.spacing.unit * 1.25,
    "paddingLeft": theme.spacing.unit * 2.5,
    "paddingRight": theme.spacing.unit * 2.5,
    "&:hover": {
      border: "1px solid",
      borderColor: theme.palette.reaction.activeElementBorderColor,
      backgroundColor: theme.palette.primary.contrastText
    }
  },
  activeVariant: {
    border: "1px solid",
    borderColor: theme.palette.reaction.activeElementBorderColor
  }
});
