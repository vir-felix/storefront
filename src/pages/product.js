import React, { Component } from "react";
import PropTypes from "prop-types";
import withData from "lib/apollo/withData";
import withRoot from "lib/theme/withRoot";
import withShop from "containers/shop/withShop";
import withCatalogItemProduct from "containers/catalog/withCatalogItemProduct";
import Layout from "components/Layout";
import ProductDetail from "components/ProductDetail";
import withTracking from "lib/tracking/withTracking";
import trackProductViewed from "lib/tracking/trackProductViewed";

@withData
@withRoot
@withShop
@withCatalogItemProduct
@withTracking
@trackProductViewed({ dispatchOnMount: true })
class ProductDetailPage extends Component {
  static propTypes = {
    /**
     * Catalog Product item
     */
    product: PropTypes.object.isRequired
  }

  render() {
    return (
      <Layout>
        <ProductDetail product={this.props.product} />
      </Layout>
    );
  }
}

export default ProductDetailPage;
