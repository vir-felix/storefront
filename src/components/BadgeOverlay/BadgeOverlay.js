import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import { withStyles } from "@material-ui/core/styles";
import Badge from "components/Badge";
import { badgeStatus, isProductBestseller, isProductLowQuantity, BADGE_TYPES } from "lib/utils";

const styles = (theme) => ({
  badgeOverlay: {
    position: "relative"
  },
  faded: {
    opacity: "0.5"
  },
  status: {
    color: theme.palette.primary.contrastText,
    left: theme.spacing.unit,
    top: theme.spacing.unit
  },
  backorder: {
    backgroundColor: theme.palette.reaction.coolGrey
  },
  bestseller: {
    backgroundColor: theme.palette.reaction.badges.bestseller
  },
  lowInventory: {
    backgroundColor: theme.palette.reaction.coolGrey
  },
  sale: {
    backgroundColor: theme.palette.reaction.badges.sale
  },
  soldOut: {
    backgroundColor: theme.palette.reaction.coolGrey
  },
  warning: {
    backgroundColor: "transparent",
    color: theme.palette.secondary.main
  },
  alignRight: {
    right: 0
  },
  secondaryBadge: {
    color: theme.palette.reaction.coolGrey,
    right: theme.spacing.unit,
    top: theme.spacing.unit
  }
});


@withStyles(styles)
class BadgeOverlay extends Component {
  static propTypes = {
    children: PropTypes.node,
    classes: PropTypes.object,
    filterOnly: PropTypes.string,
    product: PropTypes.object.isRequired,
    shouldShowPrimaryOnly: PropTypes.bool
  };

  renderBadge = () => {
    const { classes, filterOnly, product, shouldShowPrimaryOnly } = this.props;
    const status = badgeStatus(product);

    if (!status) return null;

    const badgeClasses = classNames({
      [classes.status]: true,
      [classes.backorder]: status.type === BADGE_TYPES.BACKORDER,
      [classes.bestseller]: status.type === BADGE_TYPES.BESTSELLER,
      [classes.lowInventory]: status.type === BADGE_TYPES.LOW_QUANTITY,
      [classes.sale]: status.type === BADGE_TYPES.SALE,
      [classes.soldOut]: status.type === BADGE_TYPES.SOLD_OUT
    });

    if (filterOnly) {
      if (status.type === filterOnly) {
        return (
          <Badge badgeClasses={badgeClasses} label={status.label} />
        );
      }

      return null;
    }

    // If status is "BACKORDER" or "SOLD_OUT", only show primary badge
    if (status.type === "BACKORDER" || status.type === "SOLD_OUT" || shouldShowPrimaryOnly) {
      return (
        <Badge badgeClasses={badgeClasses} label={status.label} />
      );
    }

    // If any other status, check to see if secondary badges are needed
    return (
      <Fragment>
        <Badge badgeClasses={badgeClasses} label={status.label} />
        {this.renderSecondaryBadge(status.type)}
      </Fragment>
    );
  }

  renderSecondaryBadge = (primaryBadgeType) => {
    const { classes, product } = this.props;

    if (primaryBadgeType === "SALE") {
      if (isProductLowQuantity(product)) {
        return (
          <Badge badgeClasses={classes.secondaryBadge} label={"Low Inventory"} />
        );
      }
      if (isProductBestseller(product)) {
        return (
          <Badge badgeClasses={classes.secondaryBadge} label={"Bestseller"} />
        );
      }
    }

    if (primaryBadgeType === "LOW_QUANTITY") {
      if (isProductBestseller(product)) {
        return (
          <Badge badgeClasses={classes.secondaryBadge} label={"Bestseller"} />
        );
      }
    }

    return null;
  }

  render() {
    const { children, classes, product } = this.props;
    const status = badgeStatus(product) || {};

    const badgeOverlayClasses = classNames({
      [classes.badgeOverlay]: true,
      [classes.faded]: status.type === BADGE_TYPES.SOLD_OUT
    });

    return (
      <div className={badgeOverlayClasses}>
        {this.renderBadge()}
        {children}
      </div>
    );
  }
}

export default BadgeOverlay;
