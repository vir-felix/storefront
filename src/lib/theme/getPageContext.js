import { SheetsRegistry } from "jss";
import { createGenerateClassName } from "@material-ui/core/styles";
import theme from "./reactionTheme";

/**
 * @summary Creates and returns a page context object
 * @returns {Object} Page context object
 */
function createPageContext() {
  return {
    theme,
    // This is needed in order to deduplicate the injection of CSS in the page.
    sheetsManager: new Map(),
    // This is needed in order to inject the critical CSS.
    sheetsRegistry: new SheetsRegistry(),
    // The standard class name generator.
    generateClassName: createGenerateClassName()
  };
}

/**
 * Either creates a new or returns an existing page context object. If called from Node, creates
 * a new object every time. Otherwise creates and caches an object in the browser and
 * subsequent calls will return that.
 *
 * @summary Gets a page context object
 * @returns {Object} Page context object
 */
export default function getPageContext() {
  // Make sure to create a new context for every server-side request so that data
  // isn't shared between connections (which would be bad).
  if (!process.browser) {
    return createPageContext();
  }

  // Reuse context on the client-side.
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
