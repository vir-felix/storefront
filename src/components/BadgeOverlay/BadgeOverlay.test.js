import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "@material-ui/core/styles";
import theme from "lib/theme/reactionTheme";
import product from "components/ProductDetail/__mocks__/productData.mock";
import BadgeOverlay from "./BadgeOverlay";

const soldOutProduct = Object.assign({}, product, { isSoldOut: true });
const soldOutProductWithBackorder = Object.assign({}, product, { isBackorder: true, isSoldOut: true });
const saleProduct = Object.assign({}, product, { isOnSale: true });
const saleProductWithLowInventory = Object.assign({}, product, { isOnSale: true, isLowQuantity: true });
const saleProductAndBestseller = Object.assign({}, product, { isOnSale: true, isBestseller: true });
const bestsellerProductWithLowInventory = Object.assign({}, product, { isOnSale: true, isLowQuantity: true });
const lowInventory = Object.assign({}, product, { isLowQuantity: true });

test("Product with no badges", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <BadgeOverlay product={product}/>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Product with sold out badge", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <BadgeOverlay product={soldOutProduct}/>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Product with backorder badge", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <BadgeOverlay product={soldOutProductWithBackorder}/>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Product with sale badge", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <BadgeOverlay product={saleProduct}/>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Product with sale and low inventory badges", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <BadgeOverlay product={saleProductWithLowInventory}/>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Product with sale and bestseller badges", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <BadgeOverlay product={saleProductAndBestseller}/>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Product with bestseller and low inventory badges", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <BadgeOverlay product={bestsellerProductWithLowInventory}/>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Product with low inventory badge", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <BadgeOverlay product={lowInventory}/>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
