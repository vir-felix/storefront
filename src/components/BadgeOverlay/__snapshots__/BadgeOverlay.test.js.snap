// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product with backorder badge 1`] = `
<div
  className="BadgeOverlay-badgeOverlay-1"
>
  <div
    className="Badge-badge-12 BadgeOverlay-status-3 BadgeOverlay-backorder-4"
  >
    <span
      className="Badge-labelStyle-13"
    >
      Backorder
    </span>
  </div>
</div>
`;

exports[`Product with bestseller and low inventory badges 1`] = `
<div
  className="BadgeOverlay-badgeOverlay-1"
>
  <div
    className="Badge-badge-12 BadgeOverlay-status-3 BadgeOverlay-sale-7"
  >
    <span
      className="Badge-labelStyle-13"
    >
      Sale
    </span>
  </div>
  <div
    className="Badge-badge-12 BadgeOverlay-secondaryBadge-11"
  >
    <span
      className="Badge-labelStyle-13"
    >
      Low Inventory
    </span>
  </div>
</div>
`;

exports[`Product with low inventory badge 1`] = `
<div
  className="BadgeOverlay-badgeOverlay-1"
>
  <div
    className="Badge-badge-12 BadgeOverlay-status-3 BadgeOverlay-lowInventory-6"
  >
    <span
      className="Badge-labelStyle-13"
    >
      Low Inventory
    </span>
  </div>
</div>
`;

exports[`Product with no badges 1`] = `
<div
  className="BadgeOverlay-badgeOverlay-1"
/>
`;

exports[`Product with sale and bestseller badges 1`] = `
<div
  className="BadgeOverlay-badgeOverlay-1"
>
  <div
    className="Badge-badge-12 BadgeOverlay-status-3 BadgeOverlay-sale-7"
  >
    <span
      className="Badge-labelStyle-13"
    >
      Sale
    </span>
  </div>
</div>
`;

exports[`Product with sale and low inventory badges 1`] = `
<div
  className="BadgeOverlay-badgeOverlay-1"
>
  <div
    className="Badge-badge-12 BadgeOverlay-status-3 BadgeOverlay-sale-7"
  >
    <span
      className="Badge-labelStyle-13"
    >
      Sale
    </span>
  </div>
  <div
    className="Badge-badge-12 BadgeOverlay-secondaryBadge-11"
  >
    <span
      className="Badge-labelStyle-13"
    >
      Low Inventory
    </span>
  </div>
</div>
`;

exports[`Product with sale badge 1`] = `
<div
  className="BadgeOverlay-badgeOverlay-1"
>
  <div
    className="Badge-badge-12 BadgeOverlay-status-3 BadgeOverlay-sale-7"
  >
    <span
      className="Badge-labelStyle-13"
    >
      Sale
    </span>
  </div>
</div>
`;

exports[`Product with sold out badge 1`] = `
<div
  className="BadgeOverlay-badgeOverlay-1 BadgeOverlay-faded-2"
>
  <div
    className="Badge-badge-12 BadgeOverlay-status-3 BadgeOverlay-soldOut-8"
  >
    <span
      className="Badge-labelStyle-13"
    >
      Sold Out
    </span>
  </div>
</div>
`;
